### PR TITLE
Allow less strict type system

### PR DESCRIPTION
**Pull request by RZED786**
Just wanted to fix a bug I found and also add a few features (but they may not be what you want for this project so you can leave them)

### CODE REVIEW

File: src/Parser/EvaluateExpression.h [hunk 1]
```diff
@@ -133,10 +133,6 @@ void CheckIndex(const std::vector<Token>& expr, std::size_t index, bool checkLef
 
 void CompareValues(const std::vector<Token>& expr, std::size_t index, const std::string& op)
 {
-	if (expr[index - 1].type != expr[index + 1].type) {
```
Comment (DynamicSquid) [inline src/Parser/EvaluateExpression.h:136]: I think this is needed since it prevents you from doing this:

```
"int" == int
```

**Comment by DynamicSquid** [comment]
What was the bug you encountered?

**Comment by RZED786** [comment]
If I ask if a variable is equal to a value of the same type, it says that '==' can only compare values of the same type, even though they are.

**Comment by DynamicSquid** [comment]
Oh yeah! I had the same bug as well. I fixed it though. I'll be in the `v2.1` release. Thanks for noticing!

Sorry for the late reply btw

**Pull request closed without merging**